src/components/Editor/index.js

```diff
@@ -488,6 +492,15 @@ class Editor extends PureComponent<Props, State> {
     }
 
     if (selectedSource) {
+      // auto-prettyprint the source
+      const selected = selectedSource.toJS();
+      if (
+        selected.text &&
+        !isOriginalId(selected.id) &&
+        isMinified(selected.id, selected.text)
+      ) {
+        togglePrettyPrint(selectedSource.get("id"));
+      }
```

Comment:
can we do this in `actions/sources#selectSource` we like to have the actions control the business logic